### ISSUE TITLE
Importing a non-existend module breaks incremental build pipeline

### ISSUE THREAD

**Issue by AndyOGo** [webpack/webpack-dev-middleware#170]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
I import a module, but type the wrong name. Webpack can't find it and throws. This breaks my incremental build pipeline. Webpack also seems to stop watching the file system for changes.

```sh
webpack: Compiling...
webpack building...
fatal watch error!!!
```

```js
{ [ModuleNotFoundError: Module not found: Error: Cannot resolve module 'upps-this-modules-does-not-exist' in /frontend-lib/src/js]
  name: 'ModuleNotFoundError',
  message: 'Module not found: Error: Cannot resolve module \'upps-this-modules-does-not-exist\' in /frontend-lib/src/js',
  details: 'resolve module upps-this-modules-does-not-exist in /frontend-lib/src/js\n  looking for modules in /frontend-lib/src\n    /frontend-lib/src/upps-this-modules-does-not-exist doesn\'t exist (module as directory)\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/src\n      resolve file\n        /frontend-lib/src/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/src/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/src/upps-this-modules-does-not-exist.jsx doesn\'t exist\n  looking for modules in /frontend-lib/node_modules\n    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist (module as directory)\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/node_modules\n      resolve file\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn\'t exist\n  looking for modules in /frontend-lib/node_modules\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/node_modules\n      resolve file\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn\'t exist\n    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist (module as directory)',
  missing:
   [ '/frontend-lib/src/upps-this-modules-does-not-exist',
     '/frontend-lib/src/upps-this-modules-does-not-exist',
     '/frontend-lib/src/upps-this-modules-does-not-exist.js',
     '/frontend-lib/src/upps-this-modules-does-not-exist.jsx',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist.js',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist.js',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx',
     '/frontend-lib/node_modules/upps-this-modules-does-not-exist' ],
  module:
   DependenciesBlock {
     dependencies:
      [ [Object], ... ],
     blocks: [],
     variables: [ [Object] ],
     context: '/frontend-lib/src/js',
     reasons: [ [Object], [Object] ],
     debugId: 3237,
     lastId: 242,
     id: null,
     index: null,
     index2: null,
     chunks: [],
     warnings: [],
     dependenciesWarnings: [],
     errors: [],
     dependenciesErrors: [ [Circular] ],
     request: '/frontend-lib/node_modules/happypack/loader.js?id=jsx!/frontend-lib/src/js/index.js',
     userRequest: '/frontend-lib/src/js/index.js',
     rawRequest: './src/js/index',
     parser:
      Parser {
        _plugins: [Object],
        options: undefined,
        scope: undefined,
        state: undefined,
        _currentPluginApply: undefined },
     resource: '/frontend-lib/src/js/index.js',
     loaders: [ '/frontend-lib/node_modules/happypack/loader.js?id=jsx' ],
     fileDependencies: [ '/frontend-lib/src/js/index.js' ],
     contextDependencies: [],
     error: null,
     _source:
      SourceMapSource {
        _value: '\'use strict\';\n\nObject.defineProperty(exports, "__esModule", {\n  value: true\n});\n\nvar _jquery = require(\'jquery\');\n\nvar _jquery2 = _interopRequireDefault(_jquery);\n\nrequire(\'moment/locale/de\');\n\nrequire(\'moment/locale/fr-ch\');\n\nrequire(\'moment/locale/it\');\n\nvar _moment = require(\'moment\');\n\nvar _moment2 = _interopRequireDefault(_moment);\n\nvar _svg4everybody = require(\'svg4everybody\');\n\nvar _svg4everybody2 = _interopRequireDefault(_svg4everybody);\n\nrequire(\'upps-this-modules-does-not-exist\');\n\nfunction _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }\n\nrequire(\'core-js/es6/promise\');\nrequire(\'core-js/es6/symbol\');\nrequire(\'number-to-locale-string\');\n\nrequire(\'@axa-ch/style-guide\');\nrequire(\'../fonts/mtiFontTrackingCode\');\n\n// Some of our libraries require jQuery & moment to be in global namespace.\nwindow.$ = _jquery2.default;\nwindow.jQuery = _jquery2.default;\nwindow.moment = _moment2.default;\n\n// polyfilling js string helpers\nif (!String.prototype.startsWith) {\n  String.prototype.startsWith = function (searchString, rawPosition) {\n    // eslint-disable-line no-extend-native\n    var position = rawPosition || 0;\n    return undefined.indexOf(searchString, position) === position;\n  };\n}\n\nrequire(\'./agencies\');\nrequire(\'./agency-search\');\nrequire(\'./agencymap\');\nrequire(\'./agentcard\');\nrequire(\'./claim\');\nrequire(\'./close\');\nrequire(\'./collapse\');\nrequire(\'./example\');\nrequire(\'./expander\');\n// @todo: move this file to the proper clientlib as soon as we can change forms stuff\nrequire(\'./forms\');\nrequire(\'./havarie\');\nrequire(\'./icon-teaser\');\nrequire(\'./ieshims\');\nrequire(\'./iframe\');\nrequire(\'./information-screen\');\nrequire(\'./language-switch\');\nrequire(\'./local-nav\');\nrequire(\'./mobile-navigation\');\nrequire(\'./modal2\');\nrequire(\'./mortgage-calculator\');\nrequire(\'./object-fit\');\nrequire(\'./one-two-three\');\nrequire(\'./panelsnap\');\nrequire(\'./quick-search\');\nrequire(\'./related-content\');\nrequire(\'./search\');\nrequire(\'./service-nav\');\nrequire(\'./slider\');\nrequire(\'./sticky\');\nrequire(\'./sticky-comparison-table\');\nrequire(\'./video\');\n\n// Start picture element polyfill\nrequire(\'picturefill\');\n\n// Start SVG polyfill\n(0, _svg4everybody2.default)();\n\n// General initializations\n(0, _jquery2.default)(function () {\n  (0, _jquery2.default)(\'[data-search-prefill]\').each(function (index, element) {\n    var $element = (0, _jquery2.default)(element);\n\n    (0, _jquery2.default)(\'[data-search]\').on(\'axa.search.loaded\', function (event, hasResults) {\n      (0, _jquery2.default)(\'[data-no-result]\').toggle(!hasResults);\n    });\n\n    (0, _jquery2.default)(\'[data-search]\').search(\'prefill\', {\n      term: $element.data(\'searchPrefillTerm\'),\n      filter: $element.data(\'searchPrefillFilter\'),\n      order: \'relevance\',\n      hasResults: true,\n      hasFailed: false,\n      featuredResult: $element.data(\'searchPrefillFeaturedResult\'),\n      results: $element.data(\'searchPrefillResults\')\n    });\n  });\n});\n\nexports.default = {};\n\n\nmodule.hot.accept(function (err) {\n  if (err) {\n    console.error(err);\n  }\n});\n\n ;(function register() { /* react-hot-loader/webpack */ if (process.env.NODE_ENV !== \'production\') { if (typeof __REACT_HOT_LOADER__ === \'undefined\') { return; } if (typeof module.exports === \'function\') { __REACT_HOT_LOADER__.register(module.exports, \'module.exports\', "/frontend-lib/src/js/index.js"); return; } for (var key in module.exports) { if (!Object.prototype.hasOwnProperty.call(module.exports, key)) { continue; } var namedExport = void 0; try { namedExport = module.exports[key]; } catch (err) { continue; } __REACT_HOT_LOADER__.register(namedExport, key, "/frontend-lib/src/js/index.js"); } } })();',
        _name: '/frontend-lib/node_modules/happypack/loader.js?id=jsx!/frontend-lib/src/js/index.js',
        _sourceMap: [Object],
        _originalSource: undefined,
        _innerSourceMap: undefined },
     meta: {},
     assets: {},
     built: true,
     _cachedSource: null,
     issuer: 'multi aem',
     optional: false,
     useSourceMap: true,
     building: undefined,
     buildTimestamp: 1487170904266,
     cacheable: true },
  error:
   { [Error: Cannot resolve module 'upps-this-modules-does-not-exist' in /frontend-lib/src/js]
     details: 'resolve module upps-this-modules-does-not-exist in /frontend-lib/src/js\n  looking for modules in /frontend-lib/src\n    /frontend-lib/src/upps-this-modules-does-not-exist doesn\'t exist (module as directory)\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/src\n      resolve file\n        /frontend-lib/src/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/src/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/src/upps-this-modules-does-not-exist.jsx doesn\'t exist\n  looking for modules in /frontend-lib/node_modules\n    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist (module as directory)\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/node_modules\n      resolve file\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn\'t exist\n  looking for modules in /frontend-lib/node_modules\n    resolve \'file\' upps-this-modules-does-not-exist in /frontend-lib/node_modules\n      resolve file\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn\'t exist\n        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn\'t exist\n    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn\'t exist (module as directory)',
     missing:
      [ '/frontend-lib/src/upps-this-modules-does-not-exist',
        '/frontend-lib/src/upps-this-modules-does-not-exist',
        '/frontend-lib/src/upps-this-modules-does-not-exist.js',
        '/frontend-lib/src/upps-this-modules-does-not-exist.jsx',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist.js',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist.js',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx',
        '/frontend-lib/node_modules/upps-this-modules-does-not-exist' ] },
  dependencies:
   [ CommonJsRequireDependency {
       module: null,
       request: 'upps-this-modules-does-not-exist',
       userRequest: 'upps-this-modules-does-not-exist',
       range: [Object],
       loc: [Object],
       optional: false } ],
  origin:
   DependenciesBlock {
     dependencies:
      [ [Object], ... ],
     blocks: [],
     variables: [ [Object] ],
     context: '/frontend-lib/src/js',
     reasons: [ [Object], [Object] ],
     debugId: 3237,
     lastId: 242,
     id: null,
     index: null,
     index2: null,
     chunks: [],
     warnings: [],
     dependenciesWarnings: [],
     errors: [],
     dependenciesErrors: [ [Circular] ],
     request: '/frontend-lib/node_modules/happypack/loader.js?id=jsx!/frontend-lib/src/js/index.js',
     userRequest: '/frontend-lib/src/js/index.js',
     rawRequest: './src/js/index',
     parser:
      Parser {
        _plugins: [Object],
        options: undefined,
        scope: undefined,
        state: undefined,
        _currentPluginApply: undefined },
     resource: '/frontend-lib/src/js/index.js',
     loaders: [ '/frontend-lib/node_modules/happypack/loader.js?id=jsx' ],
     fileDependencies: [ '/frontend-lib/src/js/index.js' ],
     contextDependencies: [],
     error: null,
     _source:
      SourceMapSource {
        _value: '\'use strict\';\n\nObject.defineProperty(exports, "__esModule", {\n  value: true\n});\n\nvar _jquery = require(\'jquery\');\n\nvar _jquery2 = _interopRequireDefault(_jquery);\n\nrequire(\'moment/locale/de\');\n\nrequire(\'moment/locale/fr-ch\');\n\nrequire(\'moment/locale/it\');\n\nvar _moment = require(\'moment\');\n\nvar _moment2 = _interopRequireDefault(_moment);\n\nvar _svg4everybody = require(\'svg4everybody\');\n\nvar _svg4everybody2 = _interopRequireDefault(_svg4everybody);\n\nrequire(\'upps-this-modules-does-not-exist\');\n\nfunction _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }\n\nrequire(\'core-js/es6/promise\');\nrequire(\'core-js/es6/symbol\');\nrequire(\'number-to-locale-string\');\n\nrequire(\'@axa-ch/style-guide\');\nrequire(\'../fonts/mtiFontTrackingCode\');\n\n// Some of our libraries require jQuery & moment to be in global namespace.\nwindow.$ = _jquery2.default;\nwindow.jQuery = _jquery2.default;\nwindow.moment = _moment2.default;\n\n// polyfilling js string helpers\nif (!String.prototype.startsWith) {\n  String.prototype.startsWith = function (searchString, rawPosition) {\n    // eslint-disable-line no-extend-native\n    var position = rawPosition || 0;\n    return undefined.indexOf(searchString, position) === position;\n  };\n}\n\nrequire(\'./agencies\');\nrequire(\'./agency-search\');\nrequire(\'./agencymap\');\nrequire(\'./agentcard\');\nrequire(\'./claim\');\nrequire(\'./close\');\nrequire(\'./collapse\');\nrequire(\'./example\');\nrequire(\'./expander\');\n// @todo: move this file to the proper clientlib as soon as we can change forms stuff\nrequire(\'./forms\');\nrequire(\'./havarie\');\nrequire(\'./icon-teaser\');\nrequire(\'./ieshims\');\nrequire(\'./iframe\');\nrequire(\'./information-screen\');\nrequire(\'./language-switch\');\nrequire(\'./local-nav\');\nrequire(\'./mobile-navigation\');\nrequire(\'./modal2\');\nrequire(\'./mortgage-calculator\');\nrequire(\'./object-fit\');\nrequire(\'./one-two-three\');\nrequire(\'./panelsnap\');\nrequire(\'./quick-search\');\nrequire(\'./related-content\');\nrequire(\'./search\');\nrequire(\'./service-nav\');\nrequire(\'./slider\');\nrequire(\'./sticky\');\nrequire(\'./sticky-comparison-table\');\nrequire(\'./video\');\n\n// Start picture element polyfill\nrequire(\'picturefill\');\n\n// Start SVG polyfill\n(0, _svg4everybody2.default)();\n\n// General initializations\n(0, _jquery2.default)(function () {\n  (0, _jquery2.default)(\'[data-search-prefill]\').each(function (index, element) {\n    var $element = (0, _jquery2.default)(element);\n\n    (0, _jquery2.default)(\'[data-search]\').on(\'axa.search.loaded\', function (event, hasResults) {\n      (0, _jquery2.default)(\'[data-no-result]\').toggle(!hasResults);\n    });\n\n    (0, _jquery2.default)(\'[data-search]\').search(\'prefill\', {\n      term: $element.data(\'searchPrefillTerm\'),\n      filter: $element.data(\'searchPrefillFilter\'),\n      order: \'relevance\',\n      hasResults: true,\n      hasFailed: false,\n      featuredResult: $element.data(\'searchPrefillFeaturedResult\'),\n      results: $element.data(\'searchPrefillResults\')\n    });\n  });\n});\n\nexports.default = {};\n\n\nmodule.hot.accept(function (err) {\n  if (err) {\n    console.error(err);\n  }\n});\n\n ;(function register() { /* react-hot-loader/webpack */ if (process.env.NODE_ENV !== \'production\') { if (typeof __REACT_HOT_LOADER__ === \'undefined\') { return; } if (typeof module.exports === \'function\') { __REACT_HOT_LOADER__.register(module.exports, \'module.exports\', "/frontend-lib/src/js/index.js"); return; } for (var key in module.exports) { if (!Object.prototype.hasOwnProperty.call(module.exports, key)) { continue; } var namedExport = void 0; try { namedExport = module.exports[key]; } catch (err) { continue; } __REACT_HOT_LOADER__.register(namedExport, key, "/frontend-lib/src/js/index.js"); } } })();',
        _name: '/frontend-lib/node_modules/happypack/loader.js?id=jsx!/frontend-lib/src/js/index.js',
        _sourceMap: [Object],
        _originalSource: undefined,
        _innerSourceMap: undefined },
     meta: {},
     assets: {},
     built: true,
     _cachedSource: null,
     issuer: 'multi aem',
     optional: false,
     useSourceMap: true,
     building: undefined,
     buildTimestamp: 1487170904266,
     cacheable: true } }
```

```sh
ModuleNotFoundError: Module not found: Error: Cannot resolve module 'upps-this-modules-does-not-exist' in /frontend-lib/src/js
    at /frontend-lib/node_modules/webpack/lib/Compilation.js:229:38
    at onDoneResolving (/frontend-lib/node_modules/webpack/lib/NormalModuleFactory.js:29:20)
    at /frontend-lib/node_modules/webpack/lib/NormalModuleFactory.js:85:20
    at /frontend-lib/node_modules/async/lib/async.js:726:13
    at /frontend-lib/node_modules/async/lib/async.js:52:16
    at done (/frontend-lib/node_modules/async/lib/async.js:241:17)
    at /frontend-lib/node_modules/async/lib/async.js:44:16
    at /frontend-lib/node_modules/async/lib/async.js:723:17
    at /frontend-lib/node_modules/async/lib/async.js:167:37
    at /frontend-lib/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:24:19
resolve module upps-this-modules-does-not-exist in /frontend-lib/src/js
  looking for modules in /frontend-lib/src
    /frontend-lib/src/upps-this-modules-does-not-exist doesn't exist (module as directory)
    resolve 'file' upps-this-modules-does-not-exist in /frontend-lib/src
      resolve file
        /frontend-lib/src/upps-this-modules-does-not-exist doesn't exist
        /frontend-lib/src/upps-this-modules-does-not-exist.js doesn't exist
        /frontend-lib/src/upps-this-modules-does-not-exist.jsx doesn't exist
  looking for modules in /frontend-lib/node_modules
    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn't exist (module as directory)
    resolve 'file' upps-this-modules-does-not-exist in /frontend-lib/node_modules
      resolve file
        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn't exist
        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn't exist
        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn't exist
  looking for modules in /frontend-lib/node_modules
    resolve 'file' upps-this-modules-does-not-exist in /frontend-lib/node_modules
      resolve file
        /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn't exist
        /frontend-lib/node_modules/upps-this-modules-does-not-exist.js doesn't exist
        /frontend-lib/node_modules/upps-this-modules-does-not-exist.jsx doesn't exist
    /frontend-lib/node_modules/upps-this-modules-does-not-exist doesn't exist (module as directory)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
```js
import 'upps-this-modules-does-not-exist'
```

**What is the expected behavior?**
I expect that importing a non-existend module does not break my incremental build during development.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention your webpack and Operating System version.**
Webpack: `^1.13.2`
Mac OSX El Capitan: `10.11.6`

**Comment by AndyOGo** [comment]
Sorry I just doublechecked our webpack configs for prod and dev and it turned we had set the webpack's `bail` option to `true`...

FYI: http://webpack.github.io/docs/configuration.html#bail

**Comment by Venryx** [comment]
Since that shows high in search results on the topic, I thought I'd add the problem (and solution) for my case.

So the problem was the same as yours above, basically, in that regular compile errors were causing webpack to crash, even though `watch` was enabled, and `bail` was not enabled.

I eventually found the cause -- the following (inline) plugin was present in the project's `webpack.config.js` file:
```diff
webpackConfig.plugins = [
	// Plugin to show any webpack warnings and prevent tests from running
	function () {
		let errors = []
		this.plugin("done", function (stats) {
			if (stats.compilation.errors.length) {
				// Log each of the warnings
				stats.compilation.errors.forEach(function (error) {
					errors.push(error.message || error)
				})

				// Pretend no assets were generated. This prevents the tests from running, making it clear that there were warnings.
+				throw new Error(errors)
			}
		})
	},
```

That highlighted `throw` line was the culprit. It was included in the template project from the yoeman-generator I used to begin my app, and it lay there causing the incremental builds (after an error) to not work.

Anyway, to fix it, all I did was comment out the `throw` line -- it's only useful if you're doing test-runs anyway, which I'm not. (here is the issue I opened for it, if needed for reference: https://github.com/prescottprue/generator-react-firebase/issues/36)

Anyway, just thought I'd mention it for anyone else stumbling upon this thread while searching, for a similar problem.
